common/lib/transport/connectionmanager.js

```diff
@@ -680,6 +682,14 @@ var ConnectionManager = (function() {
 		this.unpersistConnection();
 	};
 
+	ConnectionManager.prototype.checkConnectionStateFreshness = function() {
+		var sinceLast = Utils.now() - this.lastActivity;
+		if(sinceLast > this.connectionStateTtl) {
```

Comment:
Shouldn't this be `connectionStateTtl + maxIdleInterval` though as there is a small window in which the no update may have been received by the client but the connection is still active?